### ISSUE TITLE
[possible bug] summarize_by_organism cannot find sample table

### ISSUE THREAD

**Issue by bkellman** [matsengrp/phip-flow#68]
I am currently running:
```
$ nextflow run matsengrp/phip-flow -r V1.11  \
        --sample_table ../data-raw/20230919_virscan_demo.samples.debug.csv \
        --peptide_table ../data-raw/VIR3_clean.csv \
        --read_length 101 --peptide_tile_length 200 \
        --run_zscore_fit_predict \
        --run_cpm_enr_workflow \
        --summarize_by_organism true \
        --peptide_org_col Organism \
        --results 20230919_virscan_demo_phipflow_"$(date -I)" \
        -resume

N E X T F L O W  ~  version 23.10.0
Launching `https://github.com/matsengrp/phip-flow` [cheeky_wescoff] DSL2 - revision: a60f8ebf29 [V1.11]
P H I P - F L O W!
Matsen, Overbaugh, and Minot Labs
Fred Hutchinson CRC, Seattle WA
================================
sample_table    : ../data-raw/20230919_virscan_demo.samples.debug.csv
peptide_table   : ../data-raw/VIR3_clean.csv
results         : 20230919_virscan_demo_phipflow_2023-10-31

executor >  local (1)
[3a/a4e2cb] process > ALIGN:validate_sample_table (1)                            [100%] 1 of 1, cached: 1 ✔
[4f/2c6603] process > ALIGN:validate_peptide_table (1)                           [100%] 1 of 1, cached: 1 ✔
[a9/249fba] process > ALIGN:generate_fasta_reference (1)                         [100%] 1 of 1, cached: 1 ✔
[43/2583fe] process > ALIGN:generate_index (1)                                   [100%] 1 of 1, cached: 1 ✔
[94/82af45] process > ALIGN:short_read_alignment (24)                            [100%] 26 of 26, cached: 26 ✔
[07/ed5cb7] process > ALIGN:sam_to_counts (24)                                   [100%] 26 of 26, cached: 26 ✔
[35/765f97] process > ALIGN:sam_to_stats (23)                                    [100%] 26 of 26, cached: 26 ✔
[f9/4fd8f0] process > ALIGN:collect_phip_data (1)                                [100%] 1 of 1, cached: 1 ✔
[9f/ec1433] process > ALIGN:replicate_counts (1)                                 [100%] 1 of 1, cached: 1 ✔
[57/6a8991] process > STATS:counts_per_million (1)                               [100%] 1 of 1, cached: 1 ✔
[be/a7ec54] process > STATS:size_factors (1)                                     [100%] 1 of 1, cached: 1 ✔
[ce/5c30dd] process > STATS:edgeR_BEER_workflows:to_csv (1)                      [100%] 1 of 1, cached: 1 ✔
[f1/bb5bd8] process > STATS:edgeR_BEER_workflows:run_edgeR (1)                   [100%] 1 of 1, cached: 1 ✔
[85/a5caa9] process > STATS:edgeR_BEER_workflows:append_assay_csvs_to_xarray (1) [100%] 1 of 1, cached: 1 ✔
[c1/a530b7] process > STATS:edgeR_BEER_workflows:publish_rds (1)                 [100%] 1 of 1, cached: 1 ✔
[a9/6d470e] process > STATS:cpm_fold_enrichment (1)                              [100%] 1 of 1, cached: 1 ✔
[58/301d3b] process > STATS:fit_predict_zscore (1)                               [100%] 1 of 1, cached: 1 ✔
[99/68f470] process > STATS:merge_binary_datasets                                [100%] 1 of 1, cached: 1 ✔
[a7/20c77b] process > DSOUT:dump_binary                                          [100%] 1 of 1, cached: 1 ✔
[bd/a239b9] process > DSOUT:dump_wide_csv                                        [100%] 1 of 1, cached: 1 ✔
[-        ] process > DSOUT:dump_tall_csv                                        -
[af/456d3e] process > AGG:split_samples                                          [  0%] 0 of 1
[-        ] process > AGG:aggregate_organisms                                    -
[-        ] process > AGG:join_organisms                                         -
ERROR ~ Error executing process > 'AGG:split_samples'

Caused by:
  Process `AGG:split_samples` terminated with an error exit status (1)

Command executed [/home/bek321/.nextflow/assets/matsengrp/phip-flow/templates/split_samples.py]:

  #!/usr/bin/env python3

  from collections import defaultdict
  import os
  from typing import List
  import pandas as pd
  import logging


  def setup_logging() -> logging.Logger:
      """Set up logging."""

      # Set up logging
      logFormatter = logging.Formatter(
          '%(asctime)s %(levelname)-8s [split_samples] %(message)s'
      )
      logger = logging.getLogger()
      logger.setLevel(logging.INFO)

      # Also write to STDOUT
      consoleHandler = logging.StreamHandler()
      consoleHandler.setFormatter(logFormatter)
      logger.addHandler(consoleHandler)

      return logger

  logger = setup_logging()

  # The user must specify a CSV containing the sample mapping
  sample_mapping_fp = "data_sample_annotation_table.csv"
  logger.info(f"Reading in sample mapping from: {sample_mapping_fp}")
  assert os.path.exists(sample_mapping_fp)

  # Read in the table
  df = pd.read_csv(sample_mapping_fp, index_col=0)
  logger.info(f"Sample mapping table has {df.shape[0]:,} rows and {df.shape[1]:,} columns")

  # The user must specify the column used to group replicates
  # from the same sample
  sample_grouping_col = ""

  msg = f"Column '{sample_grouping_col}' not found ({', '.join(df.columns.values)})"
  assert sample_grouping_col in df.columns.values, msg

  # Write out a file containing the unique list of sample names
  df.reindex(
      columns=[sample_grouping_col]
  ).drop_duplicates(
  ).to_csv(
      "sample_list",
      header=None,
      index=None
  )

Command exit status:
  1

Command output:
  (empty)

Command error:
  2023-10-31 23:27:39,491 INFO     [split_samples] Reading in sample mapping from: data_sample_annotation_table.csv
  Traceback (most recent call last):
    File ".command.sh", line 32, in <module>
      assert os.path.exists(sample_mapping_fp)
  AssertionError

Work dir:
  /n/scratch3/users/b/bek321/virscan_hipc/data-final/work/af/456d3e3af0b2ad7c60707f28466a6e

Tip: when you have fixed the problem you can continue the execution adding the option `-resume` to the run command line

 -- Check '.nextflow.log' file for details
```

The error seems to suggest that the sample table does not exist but navigating to the working directory, the link appears to be active and the file exists (see head below)

```
(phipflow) [bek321@compute-e-16-231 456d3e3af0b2ad7c60707f28466a6e]$ ls -lhat
total 592K
-rw-rw-r-- 1 bek321 bek321    1 Oct 31 23:27 .exitcode
drwxrwxr-x 2 bek321 bek321  626 Oct 31 23:27 .
-rw-rw-r-- 1 bek321 bek321  342 Oct 31 23:27 .command.err
-rw-rw-r-- 1 bek321 bek321  342 Oct 31 23:27 .command.log
-rw-rw-r-- 1 bek321 bek321    0 Oct 31 23:27 .command.out
lrwxrwxrwx 1 bek321 bek321  108 Oct 31 23:27 data_zscore.csv.gz -> /n/scratch3/users/b/bek321/virscan_hipc/data-final/work/bd/a239b981851083b8ceab677cf2c224/data_zscore.csv.gz
lrwxrwxrwx 1 bek321 bek321  114 Oct 31 23:27 data_size_factors.csv.gz -> /n/scratch3/users/b/bek321/virscan_hipc/data-final/work/bd/a239b981851083b8ceab677cf2c224/data_size_factors.csv.gz
lrwxrwxrwx 1 bek321 bek321  125 Oct 31 23:27 data_sample_annotation_table.csv.gz -> /n/scratch3/users/b/bek321/virscan_hipc/data-final/work/bd/a239b981851083b8ceab677cf2c224/data_sample_annotation_table.csv.gz
lrwxrwxrwx 1 bek321 bek321  126 Oct 31 23:27 data_peptide_annotation_table.csv.gz -> /n/scratch3/users/b/bek321/virscan_hipc/data-final/work/bd/a239b981851083b8ceab677cf2c224/data_peptide_annotation_table.csv.gz
lrwxrwxrwx 1 bek321 bek321  112 Oct 31 23:27 data_enrichment.csv.gz -> /n/scratch3/users/b/bek321/virscan_hipc/data-final/work/bd/a239b981851083b8ceab677cf2c224/data_enrichment.csv.gz
lrwxrwxrwx 1 bek321 bek321  115 Oct 31 23:27 data_edgeR_logpval.csv.gz -> /n/scratch3/users/b/bek321/virscan_hipc/data-final/work/bd/a239b981851083b8ceab677cf2c224/data_edgeR_logpval.csv.gz
lrwxrwxrwx 1 bek321 bek321  113 Oct 31 23:27 data_edgeR_logfc.csv.gz -> /n/scratch3/users/b/bek321/virscan_hipc/data-final/work/bd/a239b981851083b8ceab677cf2c224/data_edgeR_logfc.csv.gz
lrwxrwxrwx 1 bek321 bek321  112 Oct 31 23:27 data_edgeR_hits.csv.gz -> /n/scratch3/users/b/bek321/virscan_hipc/data-final/work/bd/a239b981851083b8ceab677cf2c224/data_edgeR_hits.csv.gz
lrwxrwxrwx 1 bek321 bek321  105 Oct 31 23:27 data_cpm.csv.gz -> /n/scratch3/users/b/bek321/virscan_hipc/data-final/work/bd/a239b981851083b8ceab677cf2c224/data_cpm.csv.gz
lrwxrwxrwx 1 bek321 bek321  108 Oct 31 23:27 data_counts.csv.gz -> /n/scratch3/users/b/bek321/virscan_hipc/data-final/work/bd/a239b981851083b8ceab677cf2c224/data_counts.csv.gz
-rw-rw-r-- 1 bek321 bek321    0 Oct 31 23:27 .command.begin
-rw-rw-r-- 1 bek321 bek321 1.4K Oct 31 23:27 .command.sh
-rw-rw-r-- 1 bek321 bek321 4.6K Oct 31 23:27 .command.run
drwxrwxr-x 7 bek321 bek321  240 Oct 31 23:27 ..

(phipflow) [bek321@compute-e-16-231 456d3e3af0b2ad7c60707f28466a6e]$ zcat data_sample_annotation_table.csv.gz | head
sample_id,replicate_id,control_status,sample_ID,sample_name,fastq_filepath,raw_total_sequences,reads_mapped,error_rate,average_quality,percent_mapped,percent_peptides_detected,percent_peptides_between_10_and_100
0,97,library,S1,c1,../data-raw/fastq//IDX097-c1-1_S1_R1_001.fastq.gz,10552,1268,0.002811007,31.1,12.016679302501895,0.7656502179218289,0.0031187381585410547
1,98,beads_only,S2,c2,../data-raw/fastq//IDX098-c2-1_S2_R1_001.fastq.gz,14031,1911,0.002440275,31.6,13.61984177891811,0.9769447281629852,0.007017160856717372
2,100,beads_only,S4,c4,../data-raw/fastq//IDX100-c4-1_S4_R1_001.fastq.gz,202207,28633,0.001710275,33.5,14.16024173248206,6.888512907677553,0.28380517242723596
3,101,beads_only,S5,c5,../data-raw/fastq//IDX101-c5-1_S5_R1_001.fastq.gz,11258,1596,0.002146455,32.4,14.176585539172143,0.8677888926140483,0.0031187381585410547
4,102,beads_only,S6,c6,../data-raw/fastq//IDX102-c6-1_S6_R1_001.fastq.gz,6102,857,0.001756068,33.0,14.044575549000326,0.4545560866073587,0.0015593690792705273
5,103,empirical,S7,v1,../data-raw/fastq//IDX103-v1-1_S7_R1_001.fastq.gz,4370,656,0.001841343,33.0,15.011441647597254,0.36411268000966807,0.0023390536189057906
6,104,empirical,S8,v2,../data-raw/fastq//IDX104-v2-1_S8_R1_001.fastq.gz,4785,745,0.00162137,33.5,15.569487983281086,0.3999781688328903,0.0031187381585410547
7,105,library,S9,c1,../data-raw/fastq//IDX105-c1-2_S9_R1_001.fastq.gz,74117,10804,0.001795265,33.4,14.576952655935886,3.8212339287524264,0.025729589807963697
8,107,beads_only,S11,c3,../data-raw/fastq//IDX107-c3-2_S11_R1_001.fastq.gz,4591,698,0.001687991,33.1,15.203659333478544,0.36333299547003284,0.0031187381585410547

```

Any idea what may be going on here?

**Comment by jgallowa07** [comment]
Ah yes, this seems to fail because It's looking for the non-gzipped "data_sample_annotation_table.csv". This is definitely a bug that I'll address. 

For the time being you could try running the `V1.10` version of the pipeline before this gzip bug was introduced.

**Comment by bkellman** [comment]
thanks for the quick response!
I tried running with v1.10 but the error looks the same to my eye. I also tried rerunning everything without ```-resume```

```
N E X T F L O W  ~  version 23.10.0
Launching `https://github.com/matsengrp/phip-flow` [tiny_edison] DSL2 - revision: 6015d420a2 [V1.10]
P H I P - F L O W!
Matsen, Overbaugh, and Minot Labs
Fred Hutchinson CRC, Seattle WA
================================
sample_table    : ../data-raw/20230919_virscan_demo.samples.debug.csv
peptide_table   : ../data-raw/VIR3_clean.csv
results         : 20230919_virscan_demo_phipflow_2023-11-01


executor >  local (1)
[3a/a4e2cb] process > ALIGN:validate_sample_table (1)                            [100%] 1 of 1, cached: 1 ✔
[4f/2c6603] process > ALIGN:validate_peptide_table (1)                           [100%] 1 of 1, cached: 1 ✔
[a9/249fba] process > ALIGN:generate_fasta_reference (1)                         [100%] 1 of 1, cached: 1 ✔
[43/2583fe] process > ALIGN:generate_index (1)                                   [100%] 1 of 1, cached: 1 ✔
[cd/444cbe] process > ALIGN:short_read_alignment (23)                            [100%] 26 of 26, cached: 26 ✔
[93/106a1b] process > ALIGN:sam_to_counts (25)                                   [100%] 26 of 26, cached: 26 ✔
[04/e6c74d] process > ALIGN:sam_to_stats (25)                                    [100%] 26 of 26, cached: 26 ✔
[f9/4fd8f0] process > ALIGN:collect_phip_data (1)                                [100%] 1 of 1, cached: 1 ✔
[9f/ec1433] process > ALIGN:replicate_counts (1)                                 [100%] 1 of 1, cached: 1 ✔
[57/6a8991] process > STATS:counts_per_million (1)                               [100%] 1 of 1, cached: 1 ✔
[be/a7ec54] process > STATS:size_factors (1)                                     [100%] 1 of 1, cached: 1 ✔
[ce/5c30dd] process > STATS:edgeR_BEER_workflows:to_csv (1)                      [100%] 1 of 1, cached: 1 ✔
[f1/bb5bd8] process > STATS:edgeR_BEER_workflows:run_edgeR (1)                   [100%] 1 of 1, cached: 1 ✔
[85/a5caa9] process > STATS:edgeR_BEER_workflows:append_assay_csvs_to_xarray (1) [100%] 1 of 1, cached: 1 ✔
[c1/a530b7] process > STATS:edgeR_BEER_workflows:publish_rds (1)                 [100%] 1 of 1, cached: 1 ✔
[a9/6d470e] process > STATS:cpm_fold_enrichment (1)                              [100%] 1 of 1, cached: 1 ✔
[58/301d3b] process > STATS:fit_predict_zscore (1)                               [100%] 1 of 1, cached: 1 ✔
[99/68f470] process > STATS:merge_binary_datasets                                [100%] 1 of 1, cached: 1 ✔
[a7/20c77b] process > DSOUT:dump_binary                                          [100%] 1 of 1, cached: 1 ✔
[2e/53a8f2] process > DSOUT:dump_wide_csv                                        [100%] 1 of 1, cached: 1 ✔
[-        ] process > DSOUT:dump_tall_csv                                        -
[68/3f283c] process > AGG:split_samples                                          [100%] 1 of 1, failed: 1 ✘
[-        ] process > AGG:aggregate_organisms                                    -
[-        ] process > AGG:join_organisms                                         [  0%] 0 of 1
ERROR ~ Error executing process > 'AGG:split_samples'

Caused by:
  Process `AGG:split_samples` terminated with an error exit status (1)

Command executed [/home/bek321/.nextflow/assets/matsengrp/phip-flow/templates/split_samples.py]:

 #!/usr/bin/env python3

  from collections import defaultdict
  import os
  from typing import List
  import pandas as pd
  import logging


  def setup_logging() -> logging.Logger:
      """Set up logging."""

      # Set up logging
      logFormatter = logging.Formatter(
          '%(asctime)s %(levelname)-8s [split_samples] %(message)s'
      )
      logger = logging.getLogger()
      logger.setLevel(logging.INFO)

      # Also write to STDOUT
      consoleHandler = logging.StreamHandler()
      consoleHandler.setFormatter(logFormatter)
      logger.addHandler(consoleHandler)

      return logger

  logger = setup_logging()

  # The user must specify a CSV containing the sample mapping
  sample_mapping_fp = "data_sample_annotation_table.csv"
  logger.info(f"Reading in sample mapping from: {sample_mapping_fp}")
  assert os.path.exists(sample_mapping_fp)

  # Read in the table
  df = pd.read_csv(sample_mapping_fp, index_col=0)
  logger.info(f"Sample mapping table has {df.shape[0]:,} rows and {df.shape[1]:,} columns")

 # The user must specify the column used to group replicates
  # from the same sample
  sample_grouping_col = ""

  msg = f"Column '{sample_grouping_col}' not found ({', '.join(df.columns.values)})"
  assert sample_grouping_col in df.columns.values, msg

  # Write out a file containing the unique list of sample names
  df.reindex(
      columns=[sample_grouping_col]
  ).drop_duplicates(
  ).to_csv(
      "sample_list",
      header=None,
      index=None
  )

Command exit status:
  1

Command output:
  (empty)

Command error:
  2023-11-01 15:23:17,812 INFO     [split_samples] Reading in sample mapping from: data_sample_annotation_table.csv
  2023-11-01 15:23:17,815 INFO     [split_samples] Sample mapping table has 26 rows and 12 columns
  Traceback (most recent call last):
    File ".command.sh", line 43, in <module>
      assert sample_grouping_col in df.columns.values, msg
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  AssertionError: Column '' not found (replicate_id, control_status, sample_ID, sample_name, fastq_filepath, raw_total_sequences, reads_mapped, error_rate, average_quality, percent_mapped, percent_peptides_detected, percent_peptides_between_10_and_100)

Work dir:
  /n/scratch3/users/b/bek321/virscan_hipc/data-final/work/68/3f283c4c994c0f23c1dfa6b4be77e3

Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`

 -- Check '.nextflow.log' file for details

```

**Comment by bkellman** [comment]
@jgallowa07 It looks like I misspoke. V1.10 gives a different error. It appears to be looking for an unnamed column. Is this resolvable in the interim?

**Comment by jgallowa07** [comment]
It looks like it's looking for an empty column. Maybe double check that you specify all of the col parameters listed [here](https://matsengrp.github.io/phippery/alignments-pipeline.html#virscan-organism-summary) correctly?

**Comment by bkellman** [comment]
Hey @jgallowa07! Great catch! It looks like there's an issue with the internal sample annotation file. The columns noted by the error are from ```data_sample_annotation_table.csv```, it actually looks like v1.10 crashes on split_samples

```
  AssertionError: Column '' not found (replicate_id, control_status, sample_ID, sample_name, fastq_filepath, raw_total_sequences, reads_mapped, error_rate, average_quality, percent_mapped, percent_peptides_detected, percent_peptides_between_10_and_100)

Work dir:
  /n/scratch3/users/b/bek321/virscan_hipc/data-final/work/68/3f283c4c994c0f23c1dfa6b4be77e3


Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`
Nov-05 22:13:29.881 [Task monitor] DEBUG nextflow.Session - Session aborted -- Cause: Process `AGG:split_samples` terminated with an error exit status (1)
Nov-05 22:13:29.905 [Task submitter] DEBUG n.processor.TaskPollingMonitor - %% executor local > tasks in the submission queue: 1 -- tasks to be submitted are shown below
~> TaskHandler[id: 98; name: AGG:join_organisms; status: NEW; exit: -; error: -; workDir: /n/scratch3/users/b/bek321/virscan_hipc/data-final/work/48/7092a82541ac741050daee08c7bf25]
Nov-05 22:13:29.905 [Task monitor] DEBUG nextflow.Session - The following nodes are still active:
[process] AGG:join_organisms
  status=ACTIVE
  port 0: (value) bound ; channel: input/
  port 1: (cntrl) -     ; channel: $

Nov-05 22:13:29.907 [main] DEBUG nextflow.Session - Session await > all processes finished
Nov-05 22:13:29.907 [main] DEBUG nextflow.Session - Session await > all barriers passed
Nov-05 22:13:29.908 [Task monitor] DEBUG n.processor.TaskPollingMonitor - <<< barrier arrives (monitor: local) - terminating tasks monitor poll loop
Nov-05 22:13:29.920 [main] DEBUG n.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=95; failedCount=1; ignoredCount=0; cachedCount=0; pendingCount=1; submittedCount=0; runningCount=0; retriesCount=0; abortedCount=0; succeedDuration=59m 50s; failedDuration=721ms; cachedDuration=0ms;loadCpus=0; loadMemory=0; peakRunning=15; peakCpus=15; peakMemory=0; ]
Nov-05 22:13:30.122 [main] DEBUG nextflow.cache.CacheDB - Closing CacheDB done
Nov-05 22:13:30.150 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye

```

When I access that file in the working directory, you can see that there's a phantom first column, it looks R added rownames to a write.csv:

```
sample_id,replicate_id,control_status,sample_ID,sample_name,fastq_filepath,raw_total_sequences,reads_mapped,error_rate,average_quality,percent_mapped,percent_peptides_detected,percent_peptides_between_10_and_100
0,97,library,S1,c1,../data-raw/fastq//IDX097-c1-1_S1_R1_001.fastq.gz,10552,1268,0.002811007,31.1,12.016679302501895,0.7656502179218289,0.0031187381585410547
1,98,beads_only,S2,c2,../data-raw/fastq//IDX098-c2-1_S2_R1_001.fastq.gz,14031,1911,0.002440275,31.6,13.61984177891811,0.9769447281629852,0.007017160856717372
2,100,beads_only,S4,c4,../data-raw/fastq//IDX100-c4-1_S4_R1_001.fastq.gz,202207,28633,0.001710275,33.5,14.16024173248206,6.888512907677553,0.28380517242723596
3,101,beads_only,S5,c5,../data-raw/fastq//IDX101-c5-1_S5_R1_001.fastq.gz,11258,1596,0.002146455,32.4,14.176585539172143,0.8677888926140483,0.0031187381585410547
4,102,beads_only,S6,c6,../data-raw/fastq//IDX102-c6-1_S6_R1_001.fastq.gz,6102,857,0.001756068,33.0,14.044575549000326,0.4545560866073587,0.0015593690792705273
5,103,empirical,S7,v1,../data-raw/fastq//IDX103-v1-1_S7_R1_001.fastq.gz,4370,656,0.001841343,33.0,15.011441647597254,0.36411268000966807,0.0023390536189057906

```

**Comment by jgallowa07** [comment]
The "phantom" column you're referring to is the "sample_id" - this column is described in the [documentation here](https://matsengrp.github.io/phippery/alignments-pipeline.html#sample-table):
```
One caveat is that you should not include a column named sample_id, this feature name is reserved for integer id’s which are added to both the sample and peptide annotation tables for internal and user organization of the annotations with enrichment tables. 
```
So this is to be expected. However this is not the reason for the crashing.

I'm assuming you're running the same command as the first post, but with V1.10? i.e.
```
$ nextflow run matsengrp/phip-flow -r V1.10  \
        --sample_table ../data-raw/20230919_virscan_demo.samples.debug.csv \
        --peptide_table ../data-raw/VIR3_clean.csv \
        --read_length 101 --peptide_tile_length 200 \
        --run_zscore_fit_predict \
        --run_cpm_enr_workflow \
        --summarize_by_organism true \
        --peptide_org_col Organism \
        --results 20230919_virscan_demo_phipflow_"$(date -I)" \
        -resume
```
correct?

If so, I assume it's crashing because the necessary parameter `--sample_grouping_col` was not specified. More specifically
in your case, because the parameter was [not specified, the code ](https://github.com/matsengrp/phip-flow/blob/6015d420a281ac521430de0d136afbfba336c5a7/templates/split_samples.py#L40) uses the [default value](https://github.com/matsengrp/phip-flow/blob/6015d420a281ac521430de0d136afbfba336c5a7/nextflow.config#L93) `""` (empty string). This of course is not a valid column, thus it throws the error you're seeing.

In the case you would not like to aggregate replicates (i.e. report the results independently for each row in the sample table) - you need only to specify a column which has a unique value for all samples. For instance, it looks like your "replicate_id" column is already


This does not seem like a reasonable default - and we should change that. @sminot - do you think

This parameter specifies how you would like to group aggregate replicates in the sample table. If you do not want to aggregate replicates, and have a unique identifier column (e.g. your "sample_name" column?). you could do:
```
$ nextflow run matsengrp/phip-flow -r V1.10  \
        --sample_table ../data-raw/20230919_virscan_demo.samples.debug.csv \
        --peptide_table ../data-raw/VIR3_clean.csv \
        --read_length 101 --peptide_tile_length 200 \
        --run_zscore_fit_predict \
        --run_cpm_enr_workflow \
        --summarize_by_organism true \
        --peptide_org_col Organism \
        --sample_grouping_col sample_name \                                  # <----- specify sample grouping column
        --results 20230919_virscan_demo_phipflow_"$(date -I)" \
        -resume
```
Could you try that?

**Comment by jgallowa07** [comment]
@sminot In the case where users would not like to group aggregate samples and thus do not provide `--sample_grouping_col` results in the error as described above. I could totally see as being a little confusing. Do you think that we might instead make the default equal to our unique column "sample_id" so that the default behavior simply does not aggregate across samples? Maybe I'm missing something ...

**Comment by sminot** [comment]
This is a good point, and I must apologize for not envisioning this scenario earlier.

When I was developing this set of features, the availability of multiple observations per biological sample was being relied upon heavily (which explains the summary metrics describing the number of concordant observations per sample).

In short, I think that making the default behavior skip any aggregation by sample would be a great idea, I approve.

**Comment by jgallowa07** [comment]
Thanks, Sam! I may need a little help here if you have a few moments. See the [PR above](https://github.com/matsengrp/phip-flow/pull/69) for details.

Sorry, @bkellman - the above advice may not suffice. This should be able to be patched pretty soon.

**Comment by jgallowa07** [comment]
@bkellman We've [merged a patch](https://github.com/matsengrp/phip-flow/pull/69) to `main` that should solve things. 

could I get you to try running your same command, but on the main branch like so:
```
$ nextflow run matsengrp/phip-flow -r main  \
        --sample_table ../data-raw/20230919_virscan_demo.samples.debug.csv \
        --peptide_table ../data-raw/VIR3_clean.csv \
        --read_length 101 --peptide_tile_length 200 \
        --run_zscore_fit_predict \
        --run_cpm_enr_workflow \
        --summarize_by_organism true \
        --peptide_org_col Organism \
        --results 20230919_virscan_demo_phipflow_"$(date -I)"
```
This should now run the organism aggregation workflow without specifying a column for aggregating samples.

**Comment by bkellman** [comment]
Thanks, this works perfectly! Will you merge this to 1.11?

**Comment by jgallowa07** [comment]
@bkellman Fantastic - this has been included in the new `V1.12` release.

Thanks again for pointing this out and following up! Closing now.
